pkg/ucp/datamodel/credential.go

```diff
@@ -100,14 +102,30 @@ type AzureCredentialProperties struct {
 	WorkloadIdentity *AzureWorkloadIdentityCredentialProperties `json:"workloadIdentity,omitempty"`
 }
 
-// AWSCredentialProperties contains ucp AWS credential properties.
-type AWSCredentialProperties struct {
+// AWSAccessKeyCredentialProperties contains ucp AWS access key credential properties.
+type AWSAccessKeyCredentialProperties struct {
 	// AccessKeyID contains aws access key for iam.
 	AccessKeyID string `json:"accessKeyId"`
 	// SecretAccessKey contains secret access key for iam.
 	SecretAccessKey string `json:"secretAccessKey,omitempty"`
 }
 
+// AWSIRSACredentialProperties contains ucp AWS IRSA credential properties.
+type AWSIRSACredentialProperties struct {
+	// RoleARN contains aws role arn for irsa.
+	RoleARN string `json:"roleARN"`
+}
+
+// AWSCredentialProperties contains ucp AWS credential properties.
+type AWSCredentialProperties struct {
+	// Kind is the kind of AWS credential.
+	Kind string `json:"kind,omitempty"`
```

Comment:
With this update, `kind` here https://github.com/radius-project/radius/blob/3924c62fdc5207087c7c3a7db37263b45fcbe66b/pkg/ucp/datamodel/credential.go#L71 seems redundant. What's the motivation to keep both? I know this is done the same way for Azure credentials too, but just went unnoticed then.

Reply:
AWSCredentialResourceProperties is a UCPResource, and has its "kind", along with storage and the actual AWS IAM.  AWSCredentialProperties is the AWS IAM representation. But since we support 2 types of IAM ( access key and irsa), the kind in this structure helps us to navigate to the right kind of credential (AccessKeyCredential versus IRSACredential). We could just check which one is not nil, but it feels neat to have a kind.

Reply:
Thanks for clarifying. Do we have a use case where we consume [UCP kind](https://github.com/radius-project/radius/blob/3924c62fdc5207087c7c3a7db37263b45fcbe66b/pkg/ucp/datamodel/credential.go#L71) in a way that switching to consuming the [inner kind on the properties](https://github.com/radius-project/radius/blob/3924c62fdc5207087c7c3a7db37263b45fcbe66b/pkg/ucp/datamodel/credential.go#L122) won't be possible or complicated?

Reply:
I am not sure I see any currently. @rynowak , @willdavsmith , is there a use case because of which we modeled kind in both AzureCredentialResourceProperties as well as AzureCredentialProperties  ?